inc/components/class-component.php

```diff
@@ -828,20 +957,24 @@ public function jsonSerialize() {
 	public function to_array(): array {
 
 		// Add the theme name to the config as Irving core expects.
-		$this->set_config( 'theme_name', self::camel_case( $this->get_theme() ) );
-		$this->set_config( 'theme_options', array_keys( $this->camel_case_keys( array_flip( $this->get_theme_options() ) ) ) );
+		$this->set_config_value( 'theme_name', self::camel_case( $this->get_theme() ) );
+		$this->set_config_value( 'theme_options', array_keys( $this->camel_case_keys( array_flip( $this->get_theme_options() ) ) ) );
+
+		// Filter out hidden config values.
+		$config = $this->get_config();
 
-		// Null any config keys where the config schema has hidden = true.
-		foreach ( $this->get_config_schema() as $key => $config_schema ) {
-			if ( $config_schema['hidden'] ) {
-				$this->set_config( $key, null );
+		foreach ( $this->get_schema() as $key => $schema ) {
+			if ( $schema['hidden'] && isset( $config[ $key ] ) ) {
+				unset( $config[ $key ] );
 			}
 		}
 
+		// Rename to an alias if one is set.
+		$name = empty( $this->get_alias() ) ? $this->get_name() : $this->get_alias();
```

Comment:
For the sake of debugging, it's a lot easier if we don't modify `name`, but instead have the `_alias` key. For example, this way we're not looking at `irving/text` when we should be debugging in `irving/post-title`.

I'm strongly in favor of leaving alias as its own key.

Reply:
I waffled on this a bit. My thinking is that it's more consistent for the front-end Irving app if all it really needs to worry about is mapping the `name` value of each component, rather than needing to worry about an alias property. For debugging, I think we could change the original name to a property value that is still passed to the front end though, like `_aliased_from` or something?

Reply:
That works for me.